app/controllers/api/storage.php

```diff
@@ -21,8 +22,106 @@
 use Appwrite\OpenSSL\OpenSSL;
 use Appwrite\Utopia\Response;
 use Utopia\Config\Config;
+use Utopia\Validator\Integer;
+use Appwrite\Database\Exception\Authorization as AuthorizationException;
+use Appwrite\Database\Exception\Structure as StructureException;
 use Utopia\Database\Query;
-use Utopia\Validator\Numeric;
+
+App::post('/v1/storage/buckets')
+    ->desc('Create storage bucket')
+    ->groups(['api', 'storage'])
+    ->label('scope', 'buckets.write')
+    ->label('event', 'storage.buckets.create')
+    ->label('sdk.auth', [APP_AUTH_TYPE_SESSION, APP_AUTH_TYPE_KEY, APP_AUTH_TYPE_JWT])
+    ->label('sdk.namespace', 'storage')
+    ->label('sdk.method', 'createBucket')
+    ->label('sdk.description', '/docs/references/storage/create-bucket.md')
+    ->label('sdk.response.code', Response::STATUS_CODE_CREATED)
+    ->label('sdk.response.type', Response::CONTENT_TYPE_JSON)
+    ->label('sdk.response.model', Response::MODEL_BUCKET)
+    ->param('name', '', new Text(128), 'Bucket name', false)
+    ->param('maximumFileSize', 0, new Integer(), 'Maximum file size supported', true)
+    ->param('allowedFileExtensions', ['*'], new ArrayList(new Text(64)), 'Allowed file extensions', true)
+    ->param('enabled', true, new Boolean(), 'Bucket enabled', true)
+    ->param('adapter', 'local', new WhiteList(['local']), 'Storage adapter', true)
+    ->param('encryption', true, new Boolean(), 'encryption is enabled', true)
+    ->param('antiVirus', true, new Boolean(), 'Virus scanning is enabled', true)
+    ->param('read', null, new ArrayList(new Text(64)), 'An array of strings with read permissions. By default only the current user is granted with read permissions. [learn more about permissions](/docs/permissions) and get a full list of available permissions.', true)
+    ->param('write', null, new ArrayList(new Text(64)), 'An array of strings with write permissions. By default only the current user is granted with write permissions. [learn more about permissions](/docs/permissions) and get a full list of available permissions.', true)
```

Comment:
Maybe read and write should be 2nd and 3rd params?

Reply:
Do we even need `read`?

Reply:
I would stick with the `collections` behavior until we decide otherwise.

Reply:
done